metadeploy/api/views.py

```diff
@@ -190,9 +190,20 @@ def filter_get_one(self, qs):
 
     def preflight_get(self, request):
         plan = get_object_or_404(Plan.objects, id=self.kwargs["pk"])
+        scratch_org_id = request.session.get("scratch_org_id", None)
+        scratch_org = None
+        if scratch_org_id:
+            scratch_org = ScratchOrgJob.objects.filter(
+                uuid=scratch_org_id, status=ScratchOrgJob.Status.complete
+            ).first()
+        if scratch_org:
+            org_id = scratch_org.org_id
+        elif request.user.is_authenticated:
+            org_id = request.user.org_id
+        else:
+            return Response("", status=status.HTTP_404_NOT_FOUND)
```

Comment:
@wlonk I am happy for your cleanup on this logic. I think it's roughly what we want, but could probably be written clearer?